# Q9 New data.

In [6]:
# Q9: New data
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# Read data
df = pd.read_csv('csgo_dataset.csv')

print(f'shape: {df.shape}')
print(f'columns: {df.columns}')

# Keep recent matches
df = df.iloc[0:15000]


# create a dataframe with team_1 and team_2 as one column
teams_df = pd.DataFrame()
teams_df['team'] = pd.concat([df['team_1'], df['team_2']])

print(f'\nshape: {teams_df.shape}')
# Nr of unique teams in the dataset
print(f'\nNr of unique teams: {teams_df.team.nunique()}')
print(f'\nNumber of teams with more than 200 matches: {teams_df.team.value_counts()[teams_df.team.value_counts() > 200].count()}')

# Remove all games (rows) with teams that have less than 200 matches in the original dataset
df = df[df.team_1.isin(teams_df.team.value_counts()[teams_df.team.value_counts() > 200].index)]
print(f'\nshape: {df.shape}')
df = df[df.team_2.isin(teams_df.team.value_counts()[teams_df.team.value_counts() > 200].index)]
print(f'\nshape: {df.shape}')

print(f'\nNr of unique teams: {df.team_2.nunique()}')

# create a dataframe with team_1 and team_2 as one column
teams_df = pd.DataFrame()
teams_df['team'] = pd.concat([df['team_1'], df['team_2']])

print(f'\nshape: {teams_df.shape}')
# Nr of unique teams in the dataset
print(f'\nNr of unique teams: {teams_df.team.nunique()}')
print(f'\nNumber of teams with more than 200 matches: {teams_df.team.value_counts()[teams_df.team.value_counts() > 10]}')


shape: (45773, 19)
columns: Index(['date', 'team_1', 'team_2', '_map', 'result_1', 'result_2',
       'map_winner', 'starting_ct', 'ct_1', 't_2', 't_1', 'ct_2', 'event_id',
       'match_id', 'rank_1', 'rank_2', 'map_wins_1', 'map_wins_2',
       'match_winner'],
      dtype='object')

shape: (30000, 1)

Nr of unique teams: 654

Number of teams with more than 200 matches: 47

shape: (6712, 19)

shape: (3401, 19)

Nr of unique teams: 47

shape: (6802, 1)

Nr of unique teams: 47

Number of teams with more than 200 matches: forZe                253
HAVU                 245
Vitality             229
Sprout               223
Tricked              219
AGO                  214
AVANGAR              209
G2                   204
Heroic               201
Spirit               197
Nemiga               188
Windigo              186
Virtus.pro           186
Liquid               182
GamerLegion          178
Movistar Riders      174
BIG                  172
fnatic               161
Copenhagen Flames    16

## Results:
 We now have a df with 3401 games, 47 teams, all with 200+ games in total, and all between [35, 253] games in the final dataset

In [7]:
# change the column map_winner values to 1 and -1
from numpy import sign
from Preprocessing_serieA_function import preprocess_serieA_no_draws as pre

# for i in range(len(df)):
#     if sign(df['result_1'].iloc[i] - df['result_2'].iloc[i]) == 0:
#         print('\nError in row: ', i)
#         print(df.iloc[i])

df = pre(df, 'csgo')
df['map_winner'] = df['map_winner'].apply(lambda x: 1 if x == 1 else -1)



shape: (3394, 4)


# CSGO dataset is now preprocessed
### Lets: go

In [ ]:
from ADF_script import ADF

